packages/angular/plugins/component-testing.ts

```diff
@@ -0,0 +1,276 @@
+
+  if (!buildTarget.project && !graph.nodes?.[buildTarget.project]?.data) {
+    throw new Error(stripIndents`Unable to find project configuration for build target. 
+    Has project? ${buildTarget.project}
+    Has project config? ${!!graph.nodes?.[buildTarget.project]?.data}`);
+  }
+
+  const fromWorkspaceRoot = relative(workspaceRoot, pathToConfig);
+  const normalizedFromWorkspaceRootPath = lstatSync(pathToConfig).isFile()
+    ? dirname(fromWorkspaceRoot)
+    : fromWorkspaceRoot;
+  const offset = offsetFromRoot(normalizedFromWorkspaceRootPath);
+  const buildContext = createExecutorContext(
+    graph,
+    graph.nodes[buildTarget.project].data.targets,
+    buildTarget.project,
+    buildTarget.target,
+    buildTarget.configuration
+  );
+
+  const buildableProjectConfig = normalizeBuildTargetOptions(
+    buildContext,
+    ctContext,
+    offset
+  );
+
+  return {
+    ...nxBaseCypressPreset(pathToConfig),
+    // NOTE: cannot use a glob pattern since it will break cypress generated tsconfig.
+    specPattern: ['**/*.cy.ts', '**/*.cy.js'],
+    devServer: {
+      // cypress uses string union type,
+      // need to use const to prevent typing to string
+      ...({
+        framework: 'angular',
+        bundler: 'webpack',
+      } as const),
+      options: {
+        projectConfig: buildableProjectConfig,
+      },
+    },
+  };
+}
+
+function getBuildableTarget(ctContext: ExecutorContext) {
+  const targets =
+    ctContext.projectGraph.nodes[ctContext.projectName].data?.targets;
+  const targetConfig = targets?.[ctContext.targetName];
+
+  if (!targetConfig) {
+    throw new Error(
+      stripIndents`Unable to find component testing target configuration in project '${
+        ctContext.projectName
+      }'.
+      Has targets? ${!!targets}
+      Has target name? ${ctContext.targetName}
+      Has ct project name? ${ctContext.projectName}
+      `
+    );
+  }
+
+  const cypressCtOptions = readTargetOptions(
+    {
+      project: ctContext.projectName,
+      target: ctContext.targetName,
+      configuration: ctContext.configurationName,
+    },
+    ctContext
+  );
+
+  if (!cypressCtOptions.devServerTarget) {
+    throw new Error(
+      `Unable to find the 'devServerTarget' executor option in the '${ctContext.targetName}' target of the '${ctContext.projectName}' project`
+    );
+  }
+
+  return parseTargetString(cypressCtOptions.devServerTarget);
+}
+
+function normalizeBuildTargetOptions(
+  buildContext: ExecutorContext,
+  ctContext: ExecutorContext,
+  offset: string
+): { root: string; sourceRoot: string; buildOptions: BrowserBuilderSchema } {
+  const options = readTargetOptions<BrowserBuilderSchema>(
+    {
+      project: buildContext.projectName,
+      target: buildContext.targetName,
+      configuration: buildContext.configurationName,
+    },
+    buildContext
+  );
+  const buildOptions = withSchemaDefaults(options);
+
+  buildOptions.polyfills = join(offset, buildOptions.polyfills);
```

Comment:
Good catch! these paths are in place of what is read via the angular builders, so should be unix paths since cypress just passes that to angular under the hood.